python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
I discovered this because cfgPath had picked up the value of `PCP_DIR` (set by the devtoolset-3 scl).  I presume that `stat()` system calls happen for every directory in that path.